pkg/agent/tunnel/tunnel.go

```diff
@@ -42,7 +42,7 @@ func getAddresses(endpoint *v1.Endpoints) []string {
 		for _, address := range subset.Addresses {
```

Comment:
Thanks for the update! I think we should default the port before the for loop, like:
```
if port == "" {
	port = "443"			
}
```